### ISSUE TITLE
Decode Error?

### ISSUE THREAD

**Issue by fu050409** [malwaredllc/byob#187]
# Installation
Dear @malwaredllc,
Before the start, I have to say that I'm a 14-year-old Chinese student. My English is not so good, excuse me.
It takes me a long time to install pycrypto but I succeeded in the end by install Visual Studio 2019 Community.
# Test
I feel happy that I can have a BotNet test. But while I run "python server.py --host 0.0.0.0 --port 3" (The server is CentOS) and  runs the compiled BYOB client on my Windows 10 computer, The server returns an error, It says "[-]Failed Connection: My Server's Public IP"
# Error
## On server
I slide my CentOS Terminal up (I uses SSH), and I saw it gives me "Session init exception: 'ascii' codec can't decode byte 0xbe in position 0: ordinal not in range(128)"
### Part 1
The I tried just to run a script with Python 3.6 on my Windows 10 computer, it gives a lot 404. I paste a part of it here (It's too long so that I cannot copy the all):
``` bash
C:\Users\MyUsername\Desktop>client.py
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama
DEBUG:__main__:Path: None
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama...
DEBUG:__main__:Trying to import 'colorama' as package from: 'http://103.45.110.84:5/colorama/__init__.py'
DEBUG:__main__:[+] Importing 'colorama'
DEBUG:__main__:[+] Ready to execute 'colorama' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.initialise
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.initialise' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.initialise...
DEBUG:__main__:Trying to import 'colorama.initialise' as package from: 'http://103.45.110.84:5/colorama/initialise/__init__.py'
INFO:__main__:[-] 'colorama.initialise' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'colorama.initialise' as module from: 'http://103.45.110.84:5/colorama/initialise.py'
DEBUG:__main__:[+] Importing 'colorama.initialise'
DEBUG:__main__:[+] Ready to execute 'colorama.initialise' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.atexit
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.atexit' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.atexit...
INFO:__main__:[+] Module "colorama.atexit" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.contextlib
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.contextlib' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.contextlib...
INFO:__main__:[+] Module "colorama.contextlib" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.sys
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.sys' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.sys...
INFO:__main__:[+] Module "colorama.sys" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ansitowin32
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ansitowin32' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ansitowin32...
DEBUG:__main__:Trying to import 'colorama.ansitowin32' as package from: 'http://103.45.110.84:5/colorama/ansitowin32/__init__.py'
INFO:__main__:[-] 'colorama.ansitowin32' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'colorama.ansitowin32' as module from: 'http://103.45.110.84:5/colorama/ansitowin32.py'
DEBUG:__main__:[+] Importing 'colorama.ansitowin32'
DEBUG:__main__:[+] Ready to execute 'colorama.ansitowin32' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.re
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.re' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.re...
INFO:__main__:[+] Module "colorama.re" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.sys
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.sys' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.sys...
INFO:__main__:[+] Module "colorama.sys" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.os
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.os' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.os...
INFO:__main__:[+] Module "colorama.os" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ansi
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ansi' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ansi...
DEBUG:__main__:Trying to import 'colorama.ansi' as package from: 'http://103.45.110.84:5/colorama/ansi/__init__.py'
INFO:__main__:[-] 'colorama.ansi' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'colorama.ansi' as module from: 'http://103.45.110.84:5/colorama/ansi.py'
DEBUG:__main__:[+] Importing 'colorama.ansi'
DEBUG:__main__:[+] Ready to execute 'colorama.ansi' code
INFO:__main__:[+] 'colorama.ansi' imported succesfully!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.winterm
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.winterm' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.winterm...
DEBUG:__main__:Trying to import 'colorama.winterm' as package from: 'http://103.45.110.84:5/colorama/winterm/__init__.py'
INFO:__main__:[-] 'colorama.winterm' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'colorama.winterm' as module from: 'http://103.45.110.84:5/colorama/winterm.py'
DEBUG:__main__:[+] Importing 'colorama.winterm'
DEBUG:__main__:[+] Ready to execute 'colorama.winterm' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.win32
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.win32' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.win32...
DEBUG:__main__:Trying to import 'colorama.win32' as package from: 'http://103.45.110.84:5/colorama/win32/__init__.py'
INFO:__main__:[-] 'colorama.win32' is not a package (HTTP Error 404: File not found)
DEBUG:__main__:[+] Trying to import 'colorama.win32' as module from: 'http://103.45.110.84:5/colorama/win32.py'
DEBUG:__main__:[+] Importing 'colorama.win32'
DEBUG:__main__:[+] Ready to execute 'colorama.win32' code
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ctypes
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ctypes' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ctypes...
INFO:__main__:[+] Module "colorama.ctypes" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ctypes
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ctypes' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ctypes...
INFO:__main__:[+] Module "colorama.ctypes" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ctypes
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ctypes' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ctypes...
INFO:__main__:[+] Module "colorama.ctypes" loaded as a top level module!
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ctypes.wintypes
DEBUG:__main__:Path: ['C:\\Python27\\lib\\ctypes']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:[-] Found locally!
C:\Python27\lib\ctypes\wintypes.py:2: RuntimeWarning: Parent module 'colorama.ctypes' not found while handling absolute import
  from ctypes import *
C:\Python27\lib\ctypes\wintypes.py:18: RuntimeWarning: Parent module 'colorama.ctypes' not found while handling absolute import
  from ctypes import _SimpleCData
DEBUG:__main__:FINDER=================
DEBUG:__main__:Searching: colorama.ctypes
DEBUG:__main__:Path: ['http://103.45.110.84:5/colorama/']
INFO:__main__:Checking if in declared remote module names...
INFO:__main__:Checking if built-in....
INFO:__main__:Checking if it is name repetition...
INFO:__main__:[+] Module/Package 'colorama.ctypes' can be loaded!
DEBUG:__main__:LOADER=================
DEBUG:__main__:Loading colorama.ctypes...
INFO:__main__:[+] Module "colorama.ctypes" loaded as a top level module!
INFO:__main__:[+] 'colorama.win32' imported succesfully!
INFO:__main__:[+] 'colorama.winterm' imported succesfully!
INFO:__main__:[+] 'colorama.ansitowin32' imported succesfully!
INFO:__main__:[+] 'colorama.initialise' imported succesfully!
INFO:__main__:[+] 'colorama' imported succesfully!
DEBUG:__main__:[*] Searching http://103.45.110.84:4
DEBUG:__main__:[+] Adding escalate
DEBUG:__main__:[+] Adding icloud
DEBUG:__main__:[+] Adding keylogger
DEBUG:__main__:[+] Adding miner
DEBUG:__main__:[+] Adding outlook
DEBUG:__main__:[+] Adding packetsniffer
DEBUG:__main__:[*] Searching http://103.45.110.84:4/payloads
DEBUG:__main__:[+] Adding persistence
DEBUG:__main__:[+] Adding phone
DEBUG:__main__:[+] Adding portscanner
DEBUG:__main__:[+] Adding process
DEBUG:__main__:[+] Adding ransom
DEBUG:__main__:[+] Adding screenshot
DEBUG:__main__:[+] Adding spreader
DEBUG:__main__:[*] Searching http://103.45.110.84:4/stagers
DEBUG:__main__:[+] Adding util
DEBUG:__main__:[+] Adding webcam
DEBUG:__main__:[*] Searching http://103.45.110.84:5
DEBUG:__main__:[*] Searching http://103.45.110.84:5/_dummy_thread
DEBUG:__main__:[*] Searching http://103.45.110.84:5/_markupbase
DEBUG:__main__:[*] Searching http://103.45.110.84:5/_thread
DEBUG:__main__:[*] Searching http://103.45.110.84:5/aliyunsdkcore
DEBUG:__main__:[*] Searching http://103.45.110.84:5/aliyunsdkkms
DEBUG:__main__:[*] Searching http://103.45.110.84:5/altgraph
DEBUG:__main__:[+] Adding appdirs
DEBUG:__main__:[+] Adding augeas
DEBUG:__main__:[*] Searching http://103.45.110.84:5/backports
DEBUG:__main__:[*] Searching http://103.45.110.84:5/blivet
DEBUG:__main__:[*] Searching http://103.45.110.84:5/builtins
DEBUG:__main__:[*] Searching http://103.45.110.84:5/certifi
DEBUG:__main__:[*] Searching http://103.45.110.84:5/chardet
DEBUG:__main__:[*] Searching http://103.45.110.84:5/click
DEBUG:__main__:[*] Searching http://103.45.110.84:5/colorama
DEBUG:__main__:[*] Searching http://103.45.110.84:5/concurrent
......
```

### Part 2 
After this, the script is just searching and adding. I haven't found any errors (Maybe exists).

``` bash
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: ''
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:recv_task socket error: SERVER DISCONNECTED GRACEFULLY - [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: [Errno 10054]
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
DEBUG:__main__:_get_resources error: <urlopen error [Errno 10061] >
```

At this time, the server has thrown the same error, so I shut down the server and did not let the client continue to run. I tried to read the source code but didn't find a suitable solution. So I gave up. Wish you can see this issue and help me.
Express my thanks in advance.

**Comment by malwaredllc** [comment]
Use Python 2.7 - the migration to Python 3+ is not completed yet.

**Comment by fu050409** [comment]
Thank you for replying, but I do use Python2.7.15.

**Comment by fu050409** [comment]
You can see paths similar to 'C:\\Python27\\lib\\ctypes' in my command line i pasted.
I seem to know why you think I'm using Python3, I'm sorry the above is an input error of mine.

**Comment by fu050409** [comment]
![clientcmd.jpg](https://qu-c.gitee.io/BYOB/clientcmd.jpg)
## Client (Windows 10, Python27)
![serverssh.jpg](https://qu-c.gitee.io/BYOB/serverssh.jpg)
## Server (CentOS 7, Python27)

**Comment by fu050409** [comment]
> Use Python 2.7 - the migration to Python 3+ is not completed yet.

I can't wait to get your reply. : )

**Comment by malwaredllc** [comment]
@fu050409 A lot of bugs were fixed in #191 and support for python 3 was added, so do a clean install and try again now. Let me know if you still have issues.
